samples/templates/express-direct-auth/web-server/routes/login.js

```diff
@@ -61,15 +51,24 @@ router.get('/login', (req, res) => {
   if (widget) {
     renderLoginWithWidget(req, res);
   } else {
-    renderLogin(req, res); 
+    renderTemplate(req, res, 'login');
   }
 });
 
 router.post('/login', async (req, res) => {
+  const authClient = getAuthClient(req);
   try {
+    // Authenticate using username and password and get tokens
     const { username, password } = req.body;
-    // Get tokens and userInfo
-    const authClient = getAuthClient(req);
+
+    // client-side validation
```

Comment:
How about let's add some client-side validation logic inside the SDK, as we have all required metadata from the idxResponse. Also, there is also another layer of input validation on the API side, I think devs can just consume the errors from the SDK instead of add by themselves.

Reply:
or an even simpler solution, we can add `required` attr to the inputs in the form.

Reply:
Client-side validation is done before the request to the server, so not sure how "metadata from the idxResponse" will help. It is tightly bound to the form, which is owned by the sample/client app.  I added this to the sample to make it easier to understand. If you post a blank form you will get a cryptic error from the backend.

Reply:
I was thinking in most IDX scenarios, the response drives the client forms, like the SDK gets `enroll-profile` remediation (with form and related fields metadata) first, then guides the client on how to render the page. In this case, the SDK knows how to validate the form inputs before reaching out to the API layer. IMO, it can a feature to include in the SDK layer.

Reply:
OK, let's talk more about this in design meeting